activestorage/lib/active_storage/validations/attached_validator.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  module Validations
+    class AttachedValidator < ActiveModel::EachValidator
+      def validate_each(record, attribute, value)
+        unless record.send(attribute).attached?
```

Comment:
`attached?` just looks to see if an associated `ActiveStorage::Attachment` record is present ([see here](https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/attached/one.rb#L46))

On creation, an attachment is only persisted/saved *after* the model itself has been persisted/saved ([see here](https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/attached/model.rb#L54)). Until then, it's simply noted in `attachment_changes`. 

So basically this validation will always fail on create because at that time a record will never have an associated blob saved. 

One (potential) way around this is too perform a different check if the record is `persisted?` or not. If not persisted then check that `attachment_changes1 at least contains a promise that an attachment *will* be saved. 

Would love if someone with more in depth knowledge could confirm the above.

Reply:
Also if true, scenarios like the above are why we need tests around all major combinations of these features 😃

Reply:
Should also use a guard statement here. 
```ruby
return if record.send(attribute).attached?
record.errors.add(attribute, :blank)
```

Reply:
why not use `value.attached?`?